runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectRunnerTest.java

```diff
@@ -87,15 +94,59 @@
  */
 @RunWith(JUnit4.class)
 public class DirectRunnerTest implements Serializable {
+  private static final Map<String, Context> TEAR_DOWN_CALLED = new HashMap<>();
+
   @Rule public transient ExpectedException thrown = ExpectedException.none();
 
+  @Rule public transient TestName testName = new TestName();
+
+  @Before
+  public void before() {
+    TEAR_DOWN_CALLED.put(testName.getMethodName(), new Context());
+  }
+
+  @After
+  public void after() {
+    TEAR_DOWN_CALLED.remove(testName.getMethodName());
+  }
+
   private Pipeline getPipeline() {
     PipelineOptions opts = PipelineOptionsFactory.create();
     opts.setRunner(DirectRunner.class);
 
     return Pipeline.create(opts);
   }
 
+  @Test
+  public void tearDownRespected() {
```

Comment:
Can you move the test to be placed after `testWaitUntilFinishTimeout`, and rename to `tearsDownFnsBeforeFinishing` or something that makes the sequencing that is being tested precise?

Reply:
Not sure I get it since there is no order until you add @FixMethodOrder but will move it

Reply:
I want this because it helps me group tests into categories when reading the code rather than for execution reasons.

Thanks!